test/activerecord_spanner_mock_server/spanner_active_record_with_mock_server_test.rb

```diff
@@ -192,6 +193,69 @@ def test_selects_one_singer_without_transaction
       end
     end
 
+    def test_update_all_on_table_with_sequence_falls_back_to_pdml
+      update_sql = "UPDATE `table_with_sequence` SET `name` = @p1 WHERE `table_with_sequence`.`id` = @p2"
+
+      mutation_limit_error = GRPC::InvalidArgument.new("The transaction contains too many mutations")
+      @mock.push_error(update_sql, mutation_limit_error)
+      @mock.put_statement_result update_sql, StatementResult.new(1)
+
+      TableWithSequence.transaction isolation: :fallback_to_pdml do
+        TableWithSequence.where(id: 1).update_all(name: "New Foo Name")
+      end
+
+      update_requests = @mock.requests.select { |req| req.is_a?(ExecuteSqlRequest) && req.sql == update_sql }
+      assert_equal 2, update_requests.length, "Should have been two attempts for the UPDATE DML"
+
+      pdml_begin_request = @mock.requests.find { |req| req.is_a?(BeginTransactionRequest) && req.options&.partitioned_dml }
```

Comment:
nit: can we also add a verification that the original transaction is rolled back before the retry attempt with PDML is executed? So verify that the server receives a Rollback request before the BeginTransaction request.